modules/redis/redis.c

```diff
@@ -246,57 +233,51 @@ redis_worker_insert(LogThreadedDestDriver *s, LogMessage *msg)
       return LTR_NOT_CONNECTED;
     }
 
-  log_template_format(self->key, msg, &self->template_options, LTZ_SEND,
-                      self->super.worker.instance.seq_num, NULL, self->key_str);
-
-  if (self->param1)
-    log_template_format(self->param1, msg, &self->template_options, LTZ_SEND,
-                        self->super.worker.instance.seq_num, NULL, self->param1_str);
-  if (self->param2)
-    log_template_format(self->param2, msg, &self->template_options, LTZ_SEND,
-                        self->super.worker.instance.seq_num, NULL, self->param2_str);
+  argv = g_malloc(argc * sizeof(char *));
+  argvlen = g_malloc(argc * sizeof(size_t));
 
   argv[0] = self->command->str;
   argvlen[0] = self->command->len;
-  argv[1] = self->key_str->str;
-  argvlen[1] = self->key_str->len;
 
-  if (self->param1)
+  for (int i = 0; i < g_list_length(self->fields); i++)
     {
-      argv[2] = self->param1_str->str;
-      argvlen[2] = self->param1_str->len;
-      argc++;
+      GString *str = g_string_sized_new(1024);
```

Comment:
I think this is another good scratch-buffer candidate.

Reply:
Thanks, done :)